mmeval/metrics/swd.py

```diff
@@ -0,0 +1,305 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import numpy as np
+from scipy import ndimage
+from typing import Any, Dict, List, Sequence
+
+from mmeval.core import BaseMetric
+
+
+class SlicedWassersteinDistance(BaseMetric):
+    """SWD (Sliced Wasserstein distance) metric. We calculate the SWD of two
+    sets of images in the following way. In every 'feed', we obtain the
+    Laplacian pyramids of every images and extract patches from the Laplacian
+    pyramids as descriptors. In 'summary', we normalize these descriptors along
+    channel, and reshape them so that we can use these descriptors to represent
+    the distribution of real/fake images. And we can calculate the sliced
+    Wasserstein distance of the real and fake descriptors as the SWD of the
+    real and fake images. Note that, as with the official implementation, we
+    multiply the result by 10 to prevent the value from being too small and to
+    facilitate comparison.
```

Comment:
```suggestion
    """SWD (Sliced Wasserstein distance) metric.
    
    We calculate the SWD of two sets of images in the following way. In every
    'feed', we obtain the Laplacian pyramids of every image and extract patches
    from the Laplacian pyramids as descriptors. In 'summary', we normalize
    these descriptors along channel, and reshape them so that we can use
    these descriptors to represent the distribution of real/fake images. And
    we can calculate the sliced Wasserstein distance of the real and fake
    descriptors as the SWD of the real and fake images. Note that, as with the
    official implementation, we multiply the result by 10 to prevent the value
    from being too small and to facilitate comparison.
```